### ISSUE TITLE
Model output returns wrong dimensions

### ISSUE THREAD

**Issue by mshehrozsajjad-curevision** [facebookresearch/playtorch#142]
### Version

0.2.2

### Problem Area

react-native-pytorch-core (core package)

### Steps to Reproduce

I am using a RCNN model and I get the wrong dimensions of tensor in the output when it should be the same as input. I am always getting 28x28 no matter what is the actual image size. Also my app always crashes if size of image is greater.

### Expected Results

output: ', '[{"dtype":"float32","shape":[4,4]},{"dtype":"int64","shape":[4]},{"dtype":"float32","shape":[4,1,1024,1024]},{"dtype":"float32","shape":[4]},{"dtype":"int64","shape":[2]}]'

### Code example, screenshot, or link to repository

output: ', '[{"dtype":"float32","shape":[4,4]},{"dtype":"int64","shape":[4]},{"dtype":"float32","shape":[4,1,28,28]},{"dtype":"float32","shape":[4]},{"dtype":"int64","shape":[2]}]'

**Comment by raedle** [comment]
@mshehrozsajjad-curevision, please share a minimal code example to reproduce the issue. TY!

**Comment by mshehrozsajjad-curevision** [comment]
You can see the code here its quite simple

`
      const image = await ImageUtil.fromBundle(imageUrl);

      // Get image width and height
     const height = image.getHeight();
     const width = image.getWidth();

     // Convert image to blob, which is a byte representation of the image
    // in the format height (H), width (W), and channels (C), or HWC for short
     const blob = media.toBlob(image);

     // Get a tensor from image the blob and also define in what format
    // the image blob is.
     let tensor = torch.fromBlob(blob, [
       height,
       width,
       3,
     ]);
     
    
     tensor = tensor.to({dtype: torch.float});

    // Rearrange the tensor shape to be [CHW]
     tensor = tensor.permute([2, 0, 1]);
    
    const segmented = await model.forward(tensor).then(
       (output) => {
      
           console.log('output: ', JSON.stringify(output));
      } );
`

**Comment by raedle** [comment]
The example code provided is not sufficient (e.g., the model is missing). Please provide a minimal reproducible example that shows the issue mentioned in the issue.

Also note, the example code uses both, the `then` function of the Promise API and the `async/await` syntax

**Comment by mshehrozsajjad-curevision** [comment]
I found some help with model. I have to resize the mask to fit the input image. I have mostly translated most of the python code to work with playtorch with the exception to the last line. Is there any way to implement that using playtorch? 

`

     def paste_mask_in_image_old(mask, box, img_h, img_w, threshold):
      """
      Paste a single mask in an image.
     This is a per-box implementation of :func:paste_masks_in_image.
     This function has larger quantization error due to incorrect pixel
     modeling and is not used anymore.
      Args:
        mask (Tensor): A tensor of shape (Hmask, Wmask) storing the mask of a single
            object instance. Values are in [0, 1].
        box (Tensor): A tensor of shape (4, ) storing the x0, y0, x1, y1 box corners
            of the object instance.
        img_h, img_w (int): Image height and width.
        threshold (float): Mask binarization threshold in [0, 1].

    Returns:
        im_mask (Tensor):
            The resized and binarized object mask pasted into the original
            image plane (a tensor of shape (img_h, img_w)).
    """
    # Conversion from continuous box coordinates to discrete pixel coordinates
    # via truncation (cast to int32). This determines which pixels to paste the
    # mask onto.
    box = box.to(dtype=torch.int32)  # Continuous to discrete coordinate conversion
    # An example (1D) box with continuous coordinates (x0=0.7, x1=4.3) will map to
    # a discrete coordinates (x0=0, x1=4). Note that box is mapped to 5 = x1 - x0 + 1
    # pixels (not x1 - x0 pixels).
    samples_w = box[2] - box[0] + 1  # Number of pixel samples, *not* geometric width
    samples_h = box[3] - box[1] + 1  # Number of pixel samples, *not* geometric height

    # Resample the mask from it's original grid to the new samples_w x samples_h grid
    mask = Image.fromarray(mask.cpu().numpy())
    mask = mask.resize((samples_w, samples_h), resample=Image.BILINEAR)
    mask = np.array(mask, copy=False)

    if threshold >= 0:
        mask = np.array(mask > threshold, dtype=np.uint8)
        mask = torch.from_numpy(mask)
    else:
        # for visualization and debugging, we also
        # allow it to return an unmodified mask
        mask = torch.from_numpy(mask * 255).to(torch.uint8)

    im_mask = torch.zeros((img_h, img_w), dtype=torch.uint8)
    x_0 = max(box[0], 0)
    x_1 = min(box[2] + 1, img_w)
    y_0 = max(box[1], 0)
    y_1 = min(box[3] + 1, img_h)

    im_mask[y_0:y_1, x_0:x_1] = mask[
        (y_0 - box[1]) : (y_1 - box[1]), (x_0 - box[0]) : (x_1 - box[0])
    ]
    return im_mask`

I am refering to this line in the function: 
`  im_mask[y_0:y_1, x_0:x_1] = mask[
        (y_0 - box[1]) : (y_1 - box[1]), (x_0 - box[0]) : (x_1 - box[0])
    ]`

**Comment by raedle** [comment]
@mshehrozsajjad-curevision, it looks like the code in question replaces the values of the original image with the values of the mask that are within the bounding box corners.

It's really hard to offer support without a proper way to reproduce, but here is a take (no guarantee this works!):

```typescript
let im_mask = torch.zeros([img_h, img_w]);
const im_mask_data = im_mask.data();
const x_0 = Math.max(box[0], 0);
const x_1 = Math.min(box[2] + 1, img_w);
const y_0 = Math.max(box[1], 0);
const y_1 = Math.min(box[3] + 1, img_h);

// Translation of Python function with the only difference that data is written to im_mask_data
// instead of im_mask (setting values on Tensors is not possible in PlayTorch at this time)
//
// im_mask[y_0:y_1, x_0:x_1] = mask[
//     (y_0 - box[1]) : (y_1 - box[1]), (x_0 - box[0]) : (x_1 - box[0])
// ]
for (let y = y_0; y < y_1; y++) {
  for (let x = x_0; x < x_1; x++) {
    im_mask_data[y * img_w + x] = assumedMaskData[(y - box[1]) * Wmask + x - box[0]];
  }
}

// Reassign im_mask tensor with previously set data and reshape to original im_mask
// dimensions
im_mask = torch.tensor([...im_mask_data]).reshape(im_mask.shape);
```

Try it out with the PlayTorch app: https://snack.playtorch.dev/@raedle/paste-mask-in-image-old

![IMG_A9D2F5D7AA28-1](https://user-images.githubusercontent.com/489051/204932891-043feefb-db80-4412-b7b2-e1d40bf279d8.jpeg)

Hope that helps!

**Comment by mshehrozsajjad-curevision** [comment]
@raedle  Thanks for the help it works now. I would like to know the explanation of this part:
`im_mask_data[y * img_w + x] = assumedMaskData[(y - box[1]) * Wmask + x - box[0]];`
for future reference. Is there any documentation that you could point me to? How to access this array of data.

**Comment by mshehrozsajjad-curevision** [comment]
@raedle Sorry for disturbing you again, but is there a way to get a transparent tensor from this code? I get a grayscale image from this code, but need a transparent image.
